public/javascripts/application.js

```diff
@@ -1,2 +1,38 @@
-// Javascript Goes Here~
-//
+$( document ).ready(function() {
+
+  $( '#register' ).on('click',function(event) {
+    event.preventDefault();
```

Comment:
I think you need to call event.target.preventDefault().  You need to prevent the default action of the form (submit) from firing, not the button.